### PR TITLE
Replace date picker

### PR DESCRIPTION
**Pull request by nranas**
- Removed react-date-picker
- Added new basic date-picker component (still using react-select though so we still have styling issues, maybe solve react-select issues in another PR?)
- Added unit tests (jest)

Solves #32

### CODE REVIEW

File: src/components/elements/inputs/date/date-picker.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+function handleChange(onChange, date) {
+  console.log('handleChange was called with value: ' + date)
+  onChange(getDateString(date))
+}
+
+export default ({ className, errors, onCommit, property, value }) => {
+  const date = value ? new Date(value) : null
+  const onChange = handleChange.bind(undefined, onCommit)
```
Comment (hfjallemark) [inline src/components/elements/inputs/date/date-picker.js]: Do we really need this?

File: src/components/elements/inputs/date/date-picker.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import Year from './year'
+import Month from './month'
+import Day from './day'
+
+function getDateString(date) {
+  return date && date.toISOString().split('T')[0]
+}
+
+function handleChange(onChange, date) {
+  console.log('handleChange was called with value: ' + date)
```
Comment (hfjallemark) [inline src/components/elements/inputs/date/date-picker.js]: We can remove this, right?

File: src/components/elements/inputs/date/day.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import cx from 'classnames'
+import Select from 'react-select'
+import 'react-select/dist/react-select.css'
+
+export default ({ className, errors, onCommit, property, value }) => {
+  return (
```
Comment (hfjallemark) [inline src/components/elements/inputs/date/day.js]: We can do implicit return here.

File: src/components/elements/inputs/date/year.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import cx from 'classnames'
+import Select from 'react-select'
+import 'react-select/dist/react-select.css'
+
+export default ({ className, errors, onCommit, property, value }) => {
+  return (
```
Comment (hfjallemark) [inline src/components/elements/inputs/date/year.js]: Implicit return.

File: src/components/elements/inputs/date/year.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    <Select
+      className={cx(className, 'ct-year')}
+      onChange={selected => onCommit(selectYear(selected && selected.value, value))}
+      options={getYears().map(year => {
```
Comment (hfjallemark) [inline src/components/elements/inputs/date/year.js]: Implicit return.

File: src/components/elements/inputs/date/year.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+function getYear(value) {
+  return value && value.getFullYear()
+}
+
+function getYears() {
+  const years = []
+  let current = new Date().getFullYear() - 100
+  for (let i = 0; i <= 200; i++) {
```
Comment (hfjallemark) [inline src/components/elements/inputs/date/year.js]: We're doing this in a couple of places -- maybe write a simple range util like [SugarJS](http://sugarjs.com/ranges) has?

File: src/components/elements/inputs/date/date-picker.js (hunk 2 above)
Comment (nranas) [inline src/components/elements/inputs/date/date-picker.js]: Fixed

File: src/components/elements/inputs/date/date-picker.js (hunk 1 above)
Comment (nranas) [inline src/components/elements/inputs/date/date-picker.js]: Fixed

File: src/components/elements/inputs/date/day.js (hunk 3 above)
Comment (nranas) [inline src/components/elements/inputs/date/day.js]: Fixed

File: src/components/elements/inputs/date/year.js (hunk 4 above)
Comment (nranas) [inline src/components/elements/inputs/date/year.js]: Fixed

File: src/components/elements/inputs/date/year.js (hunk 5 above)
Comment (nranas) [inline src/components/elements/inputs/date/year.js]: Outdated after changing to regular select.

File: src/components/elements/inputs/date/year.js (hunk 6 above)
Comment (nranas) [inline src/components/elements/inputs/date/year.js]: Created a date util class

File: themes/default/input.css [hunk 7]
```diff
@@ -22,6 +22,11 @@
   border-color: #CC0302;
 }
 
+select:disabled {
```
Comment (hfjallemark) [inline themes/default/input.css:25]: Should this be more specific? E.g:

``` css
.ct-day:disabled,
.ct-month:disabled,
.ct-year:disabled {
   ...
}
```

Comment (nranas) [inline themes/default/input.css:25]: I actually thought we wanted this to apply to all select inputs but maybe we don't?

**Pull request merged**